jobbergate-api/jobbergate_api/apps/job_scripts/routers.py

```diff
@@ -176,6 +181,76 @@ async def job_script_get(job_script_id: int = Query(...)):
     return response
 
 
+@router.patch(
+    "/job-scripts/{job_script_id}/upload",
+    status_code=status.HTTP_204_NO_CONTENT,
+    description="Endpoint to replace a job script file.",
+    dependencies=[Depends(guard.lockdown(Permissions.JOB_SCRIPTS_EDIT))],
+)
+async def job_script_replace_file_content(
+    job_script_id: int = Query(...),
+    job_script_file: UploadFile = File(...),
+):
+    """Replace the content on a job script file."""
+    logger.debug(f"Replacing the main file from {job_script_id=}")
+
+    query = job_scripts_table.select().where(job_scripts_table.c.id == job_script_id)
+    logger.trace(f"get_query = {render_sql(query)}")
+    job_script = await database.fetch_one(query)
+
+    if not job_script:
+        message = f"JobScript with id={job_script_id} was not found."
+        logger.warning(message)
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=message)
+
+    file_manager = JobScriptFiles.file_manager_factory(job_script_id)
+    file_content = job_script_file.file.read().decode("utf-8")
+
+    for s3_path in file_manager.keys():
+        root_dir = s3_path.parts[0]
+        if root_dir == JOBSCRIPTS_MAIN_FILE_FOLDER:
+            file_manager[s3_path] = file_content
+            logger.debug(f"Success replacing the main file from {job_script_id=}")
+            return None
```

Comment:
will this return 204??

Reply:
Yes, confirmed by the unit tests

Reply:
hmmm interesting, I usually use `Response(status=204)`. Nice to know.